src/vsr.zig

```diff
@@ -76,13 +76,14 @@ pub const Zone = enum {
             assert(offset_logical < zone_size);
         }
 
-        return offset_logical + switch (zone) {
-            .superblock => 0,
-            .wal_headers => size_superblock,
-            .wal_prepares => size_superblock + size_wal_headers,
-            .client_replies => size_superblock + size_wal_headers + size_wal_prepares,
-            .grid => size_superblock + size_wal_headers + size_wal_prepares + size_client_replies,
-        };
+        var offset_ = offset_logical;
+        // TODO(Zig): "inline for" when "unable to evaluate constant expression" is fixed.
```

Comment:
Got it to compile with 

```
        comptime var offset_ = 0;
        inline for (comptime std.enums.values(Zone)) |z| {
            if (zone == z) return offset_;
            offset_ += comptime z.size() orelse 0;
        } else {
            unreachable;
        }
```

`comptime` in `for` seems like compiler bug, but `orelse 0` seems essential, at least according to my mental model of comptime

Reply:
Can you elaborate on the `orelse 0`? `size()` is `null` only for the grid, which is the last zone, so we should never add that last size, hence `.?`. (It does need `orelse 0` to compile, but I interpreted that as another compiler bug).

Reply:
Ah, that makes sense, thanks!

Reply:
```zig
        comptime var offset_zone = 0;
        inline for (comptime std.enums.values(Zone)) |z| {
            if (zone == z) return offset_zone + offset_logical;
            offset_zone += comptime z.size() orelse 0;
        } else {
            unreachable;
        }
```

this is what we want (to force zone start offset to be evaluated at compile time).

Reply:
Ooh, great catch!

Reply:
Oh I just barely missed that last comment, I'll update it again